### ISSUE TITLE
pixhawk mini IMU missing v1.11.0-beta1

### ISSUE THREAD

**Issue by dagar** [PX4/Firmware#14745]
Two reports of the 3DR/Holybro pixhawk mini IMU not starting with v1.11.0-beta, however the single unit I have seems to be working.

 - possible confusion with px4_fmu-v2 vs px4_fmu-v3 builds was verified

**Comment by dagar** [comment]
@davids5 do you have a unit you could check quickly?

**Comment by bys1123** [comment]
Here is some screenshots
![image](https://user-images.githubusercontent.com/4748411/80172815-94198f80-8620-11ea-9bcf-44c34cc0ba2f.png)
![image](https://user-images.githubusercontent.com/4748411/80172848-aabfe680-8620-11ea-9463-1eec2ac0a227.png)
![image](https://user-images.githubusercontent.com/4748411/80172858-b0b5c780-8620-11ea-8f30-971974f179d0.png)

**Comment by dagar** [comment]
Thanks @bys1123. Could you capture the output of `ver all` as well?

**Comment by bys1123** [comment]
@dagar No problem. It's lunch time in china, will reply you half hour later.

**Comment by bys1123** [comment]
![image](https://user-images.githubusercontent.com/4748411/80174275-c0370f80-8624-11ea-83be-556f38ee7ff9.png)
@dagar

**Comment by bys1123** [comment]
![image](https://user-images.githubusercontent.com/4748411/80174430-350a4980-8625-11ea-976b-ae706cda7187.png)

**Comment by dagar** [comment]
Ok, I was able to get mine stuck in the same state after upgrading (automatic reboot). After a fresh power cycle (disconnect + reconnect usb) it works.

``` Console
FMUv2 ver 0xA : Rev 0 V2M
[boot] Fault Log info File No 4 Length 3177 flags:0x01 state:1
[boot] Fault Log is Armed
sercon: Registering CDC/ACM serial driver
sercon: Successfully registered the CDC/ACM serial driver
HW arch: PX4_FMU_V3
HW type: V2M
HW version: 0x0009000A
HW revision: 0x00000000
FW git-hash: 31f3a21849e45ea5f110ddb6de2b09a630f54fc6
FW version: 1.11.0 80 (17498240)
FW git-branch: master
OS: NuttX
OS version: Release 8.2.0 (134349055)
OS git-hash: 7fffab1610d2ea08a95383febd0579e0037b6b51
Build datetime: Apr 24 2020 00:10:24
Build uri: localhost
Toolchain: GNU GCC, 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
PX4GUID: 000100000000313237313036510500470037
MCU: STM32F42x, rev. 3
[hardfault_log] Fault Log is Armed
INFO  [param] selected parameter default file /fs/mtd_params
INFO  [tune_control] Publishing standard tune 1
Board defaults: /etc/init.d/rc.board_defaults
nsh: rgbled_pwm: command not found
Board sensors: /etc/init.d/rc.board_sensors
rgbled #0 on I2C bus 2
ms5611 #0 on SPI bus 1 (devid=0x3d)
icm20608g #0 on SPI bus 1 (devid=0x3a)
Board extras: /etc/init.d/rc.board_mavlink
INFO  [mavlink] mode: Config, data rate: 800000 B/s on /dev/ttyACM0 @ 57600B
Starting Main GPS on /dev/ttyS3
Starting MAVLink on /dev/ttyS1
INFO  [mavlink] mode: Normal, data rate: 1200 B/s on /dev/ttyS1 @ 57600B
No autostart ID found
INFO  [logger] logger started (mode=all)
```

``` Console
nsh> work_queue status

Work Queue: 6 threads                      RATE        INTERVAL
|__ 1) wq:lp_default   
|   |__ 1) send_event                   30.0 Hz      33317.7 us (33333 us)
|   \__ 2) load_mon                      1.0 Hz     988905.0 us (1000000 us)
|__ 2) wq:hp_default   
|   |__ 1) tone_alarm                   10.2 Hz      98369.9 us
|   |__ 2) rc_update                     0.0 Hz          0.0 us
|   |__ 3) adc                         100.0 Hz       9998.7 us (10000 us)
|   \__ 4) battery_status              100.0 Hz       9998.6 us (10000 us)
|__ 3) wq:I2C2         
|   \__ 1) rgbled                       38.9 Hz      25734.8 us
|__ 4) wq:SPI1         
|   |__ 1) ms5611                       99.4 Hz      10063.4 us
|   \__ 2) icm20608g                   400.4 Hz       2497.7 us
|__ 5) wq:att_pos_ctrl 
|   |__ 1) sensors                     400.4 Hz       2497.6 us
|   |__ 2) vehicle_acceleration        400.4 Hz       2497.5 us
|   |__ 3) vehicle_air_data             74.5 Hz      13421.8 us
|   |__ 4) vehicle_imu                 400.4 Hz       2497.3 us
|   \__ 5) ekf2                        400.4 Hz       2497.4 us
\__ 6) wq:rate_ctrl    
    \__ 1) vehicle_angular_velocity    400.4 Hz       2497.5 us
```

**Comment by bys1123** [comment]
@dagar Tried a fresh power cycle, still not works.

**Comment by bys1123** [comment]
![image](https://user-images.githubusercontent.com/4748411/80174803-5b7cb480-8626-11ea-9998-a242abfbb51f.png)

**Comment by dagar** [comment]
I'm not seeing any issue with current master as long as px4_fmu-v3 is flashed (bootloader update may be required).

![Screenshot from 2020-06-01 14-08-26](https://user-images.githubusercontent.com/84712/83439569-6be83200-a411-11ea-907b-2b3f1ad8728a.png)

![Screenshot from 2020-06-01 14-09-08](https://user-images.githubusercontent.com/84712/83439602-7c98a800-a411-11ea-9970-1a0834a158db.png)

**Comment by bys1123** [comment]
Thanks, we will try it again.
As you see, the board is already using FMUV3 bootloader, we need to flash it again for testing, right?

**Comment by dagar** [comment]
> Thanks, we will try it again.
> As you see, the board is already using FMUV3 bootloader, we need to flash it again for testing, right?

Yes, please flash current master.

If it's still not working we'll probably need more info.
  - When did it last work? Log available?
  - Was there a board revision at some point?

**Comment by dagar** [comment]
@bys1123 if master doesn't work please try https://github.com/PX4/Firmware/pull/15013.

**Comment by bys1123** [comment]
@dagar Thanks! master doesn't work, will try #15013 tomorrow.

**Comment by davids5** [comment]
master works for me with V3 build on the device. V2 does not have the driver.

master: (rebased under my branch)

V2 Build
```
FMUv2 ver 0xA : Rev 0 V2M
[boot] Fault Log info File No 4 Length 3177 flags:0x01 state:1
[boot] Fault Log is Armed
sercon: Registering CDC/ACM serial driver
sercon: Successfully registered the CDC/ACM serial driver
HW arch: PX4_FMU_V2
HW type: V2M
HW version: 0x0009000A
HW revision: 0x00000000
FW git-hash: 72034abe943e8390bf86e256ef9e91f46f08fc32
FW version: 1.11.0 80 (17498240)
FW git-branch: master_px4_fmu-v6x
OS: NuttX
OS version: Release 8.2.0 (134349055)
OS git-hash: 81d5b634cbc890ace6ad31c798d97d5112eee21f
Build datetime: Jun  2 2020 08:32:49
Build uri: localhost
Toolchain: GNU GCC, 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
PX4GUID: 0001000000003634343035355106004e0027
MCU: STM32F42x, rev. 3
[hardfault_log] Fault Log is Armed
INFO  [param] selected parameter default file /fs/mtd_params
ERROR [parameters] ignoring unrecognised parameter 'BAT_CNT_V_CURR'
ERROR [parameters] ignoring unrecognised parameter 'BAT_CNT_V_VOLT'
INFO  [tune_control] Publishing standard tune 1
Board defaults: /etc/init.d/rc.board_defaults
  IMU_GYRO_RATEMAX: curr: 0 -> new: 400
nsh: send_event: command not found
nsh: rgbled_ncp5623c: command not found
nsh: rgbled_pwm: command not found
ERROR [param] Parameter LIGHT_EN_BLINKM not found
WARN  [px4io] check CRC failed: -22
INFO  [tune_control] Publishing standard tune 18
INFO  [px4io] default PWM output device
[PX4IO] using firmware from /etc/extras/px4_io-v2_default.bin
[PX4IO] found bootloader revision: 4
[PX4IO] erase...
[PX4IO] programming 58480 bytes...
[PX4IO] verify...
[PX4IO] update complete
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 19
Board sensors: /etc/init.d/rc.board_sensors
rgbled #0 on I2C bus 2
hmc5883 #0 on I2C bus 1 (external, equal to '-b 1')
nsh: lis3mdl: command not found
nsh: ist8310: command not found
nsh: qmc5883: command not found
nsh: rm3100: command not found
ms5611 #0 on SPI bus 1 (devid=0x3d)
nsh: icm20608g: command not found
nsh: mpu9250: command not found
ERROR [param] Parameter TRIG_MODE not found
Board extras: /etc/init.d/rc.board_mavlink
INFO  [mavlink] mode: Config, data rate: 800000 B/s on /dev/ttyACM0 @ 57600B
INFO  [mavlink] mode: Normal, data rate: 1200 B/s on /dev/ttyS1 @ 57600B
ERROR [param] Parameter DSHOT_CONFIG not found
INFO  [px4io] default PWM output device
INFO  [init] Mixer: /etc/mixers/quad_+.main.mix on /dev/pwm_output0
INFO  [init] Mixer: /etc/mixers/pass.aux.mix on /dev/pwm_output1
ERROR [param] Parameter TC_A_ENABLE not found
ERROR [param] Parameter TC_B_ENABLE not found
ERROR [param] Parameter TC_G_ENABLE not found
ERROR [param] Parameter SENS_EN_PX4FLOW not found
ERROR [param] Parameter TEL_BST_EN not found
INFO  [logger] logger started (mode=all)

NuttShell (NSH)
nsh> sensors status
INFO  [sensors] selected gyro: 0 (0)
INFO  [ecl/validation] validator: best: -1, prev best: -1, failsafe: NO (0 events)

INFO  [sensors] selected accel: 0 (0)
INFO  [ecl/validation] validator: best: -1, prev best: -1, failsafe: NO (0 events)

INFO  [sensors] selected mag: 73225 (0)
INFO  [ecl/validation] validator: best: 0, prev best: 0, failsafe: NO (0 events)
INFO  [ecl/validation] sensor #0, prio: 125, state: OK
INFO  [ecl/validation]  val:   0.0121, lp:   0.0140 mean dev:  -0.0001 RMS:   0.0031 conf:   1.0000
INFO  [ecl/validation]  val:   0.3268, lp:   0.3283 mean dev:   0.0000 RMS:   0.0021 conf:   1.0000
INFO  [ecl/validation]  val:  -0.3341, lp:  -0.3435 mean dev:   0.0001 RMS:   0.0066 conf:   1.0000

INFO  [vehicle_air_data] selected barometer: 3997706 (0)
INFO  [ecl/validation] validator: best: 0, prev best: 0, failsafe: NO (0 events)
INFO  [ecl/validation] sensor #0, prio: 75, state: OK
INFO  [ecl/validation]  val: 100515.0000, lp: 100511.7266 mean dev:   0.3196 RMS:   2.9228 conf:   1.0000
INFO  [ecl/validation]  val:  39.0500, lp:  39.0356 mean dev:   0.0491 RMS:   0.0250 conf:   1.0000
INFO  [ecl/validation]  val:   0.0000, lp:   0.0000 mean dev:   0.0000 RMS:   0.0000 conf:   1.0000

INFO  [sensors] Airspeed status:
INFO  [ecl/validation]  no data

INFO  [vehicle_acceleration] selected sensor: 0 (0), rate: 1000.0 Hz
INFO  [vehicle_acceleration] estimated bias: [0.000 0.000 0.000]

INFO  [vehicle_angular_velocity] selected sensor: 0 (0), rate: 1000.0 Hz
INFO  [vehicle_angular_velocity] estimated bias: [0.000 0.000 0.000]

nsh> work_queue status
nsh: work_queue: command not found
```
V2 Build
```
nsh>
FMUv2 ver 0xA : Rev 0 V2M
[boot] Fault Log info File No 4 Length 3177 flags:0x01 state:1
[boot] Fault Log is Armed
sercon: Registering CDC/ACM serial driver
sercon: Successfully registered the CDC/ACM serial driver
HW arch: PX4_FMU_V3
HW type: V2M
HW version: 0x0009000A
HW revision: 0x00000000
FW git-hash: 72034abe943e8390bf86e256ef9e91f46f08fc32
FW version: 1.11.0 80 (17498240)
FW git-branch: master_px4_fmu-v6x
OS: NuttX
OS version: Release 8.2.0 (134349055)
OS git-hash: 81d5b634cbc890ace6ad31c798d97d5112eee21f
Build datetime: Jun  2 2020 08:37:42
Build uri: localhost
Toolchain: GNU GCC, 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
PX4GUID: 0001000000003634343035355106004e0027
MCU: STM32F42x, rev. 3
[hardfault_log] Fault Log is Armed
INFO  [param] selected parameter default file /fs/mtd_params
INFO  [tune_control] Publishing standard tune 1
Board defaults: /etc/init.d/rc.board_defaults
nsh: rgbled_pwm: command not found
Board sensors: /etc/init.d/rc.board_sensors
rgbled #0 on I2C bus 2
hmc5883 #0 on I2C bus 1 (external, equal to '-b 1')
ms5611 #0 on SPI bus 1 (devid=0x3d)
icm20608g #0 on SPI bus 1 (devid=0x3a)
Board extras: /etc/init.d/rc.board_mavlink
INFO  [mavlink] mode: Config, data rate: 800000 B/s on /dev/ttyACM0 @ 57600B
Starting Main GPS on /dev/ttyS3
Starting MAVLink on /dev/ttyS1
INFO  [mavlink] mode: Normal, data rate: 1200 B/s on /dev/ttyS1 @ 57600B
INFO  [px4io] default PWM output device
INFO  [init] Mixer: /etc/mixers/quad_+.main.mix on /dev/pwm_output0
INFO  [init] Mixer: /etc/mixers/pass.aux.mix on /dev/pwm_output1
INFO  [logger] logger started (mode=all)

NuttShell (NSH)
nsh> INFO  [ecl/EKF] reset position to last known position
INFO  [ecl/EKF] reset velocity to zero
sensors status
INFO  [sensors] selected gyro: 3801098 (0)
INFO  [ecl/validation] validator: best: 0, prev best: 0, failsafe: NO (0 events)
INFO  [ecl/validation] sensor #0, prio: 100, state: OK
INFO  [ecl/validation]  val:   0.0047, lp:  -0.0032 mean dev:  -0.0001 RMS:   0.0027 conf:   1.0000
INFO  [ecl/validation]  val:  -0.0234, lp:  -0.0350 mean dev:  -0.0004 RMS:   0.0026 conf:   1.0000
INFO  [ecl/validation]  val:   0.0041, lp:   0.0054 mean dev:  -0.0000 RMS:   0.0011 conf:   1.0000

INFO  [sensors] selected accel: 3801098 (0)
INFO  [ecl/validation] validator: best: 0, prev best: 0, failsafe: NO (0 events)
INFO  [ecl/validation] sensor #0, prio: 100, state: OK
INFO  [ecl/validation]  val:   2.9292, lp:   2.8996 mean dev:   0.0002 RMS:   0.0151 conf:   1.0000
INFO  [ecl/validation]  val:  -3.3802, lp:  -3.3802 mean dev:  -0.0007 RMS:   0.0163 conf:   1.0000
INFO  [ecl/validation]  val:  -8.8111, lp:  -8.8923 mean dev:  -0.0035 RMS:   0.0282 conf:   1.0000

INFO  [sensors] selected mag: 73225 (0)
INFO  [ecl/validation] validator: best: 0, prev best: 0, failsafe: NO (0 events)
INFO  [ecl/validation] sensor #0, prio: 125, state: OK
INFO  [ecl/validation]  val:   0.0121, lp:   0.0126 mean dev:   0.0001 RMS:   0.0029 conf:   1.0000
INFO  [ecl/validation]  val:   0.3329, lp:   0.3296 mean dev:   0.0001 RMS:   0.0021 conf:   1.0000
INFO  [ecl/validation]  val:  -0.3439, lp:  -0.3424 mean dev:   0.0002 RMS:   0.0063 conf:   1.0000

INFO  [vehicle_air_data] selected barometer: 3997706 (0)
INFO  [ecl/validation] validator: best: 0, prev best: 0, failsafe: NO (0 events)
INFO  [ecl/validation] sensor #0, prio: 75, state: OK
INFO  [ecl/validation]  val: 100513.9922, lp: 100511.4219 mean dev:   3.4060 RMS:   4.0492 conf:   1.0000
INFO  [ecl/validation]  val:  37.9900, lp:  37.9829 mean dev:   0.0541 RMS:   0.0385 conf:   1.0000
INFO  [ecl/validation]  val:   0.0000, lp:   0.0000 mean dev:   0.0000 RMS:   0.0000 conf:   1.0000

INFO  [sensors] Airspeed status:
INFO  [ecl/validation]  no data

INFO  [vehicle_acceleration] selected sensor: 3801098 (0), rate: 400.5 Hz
INFO  [vehicle_acceleration] estimated bias: [0.000 0.000 -0.061]

INFO  [vehicle_angular_velocity] selected sensor: 3801098 (0), rate: 400.5 Hz
INFO  [vehicle_angular_velocity] estimated bias: [-0.003 -0.031 0.005]


INFO  [vehicle_imu] Accel ID: 3801098, interval: 2496.8 us, Gyro ID: 3801098, interval: 2496.8 us
vehicle_imu: accel data gap: 1 events
vehicle_imu: gyro data gap: 1 events
vehicle_imu: accel update interval: 2004 events, 2495.49us avg, min 2472us max 2522us 3.440us rms
vehicle_imu: gyro update interval: 2010 events, 2495.49us avg, min 2472us max 2522us 3.448us rms
nsh> INFO  [ecl/EKF] 9217248: EKF aligned, (baro hgt, IMU buf: 18, OBS buf: 14)

nsh> work_queue status

Work Queue: 8 threads                        RATE        INTERVAL
|__ 1) wq:lp_default
|   |__ 1) send_event                     30.0 Hz        33297 us (33333 us)
|   |__ 2) load_mon                        1.0 Hz       976183 us (1000000 us)
|   \__ 3) mc_hover_thrust_estimator       0.0 Hz            0 us
|__ 2) wq:hp_default
|   |__ 1) tone_alarm                     10.9 Hz        92005 us
|   |__ 2) rc_update                       0.0 Hz            0 us
|   |__ 3) adc                           100.0 Hz         9999 us (10000 us)
|   |__ 4) battery_status                100.0 Hz         9999 us (10000 us)
|   \__ 5) pwm_out                         0.1 Hz      5478855 us
|__ 3) wq:I2C2
|   \__ 1) rgbled                         38.6 Hz        25888 us
|__ 4) wq:I2C1
|   \__ 1) hmc5883                       125.7 Hz         7956 us
|__ 5) wq:SPI1
|   |__ 1) ms5611                         99.4 Hz        10062 us
|   \__ 2) icm20608g                     400.3 Hz         2498 us
|__ 6) wq:navigation_and_controllers
|   |__ 1) sensors                       199.6 Hz         5009 us
|   |__ 2) vehicle_acceleration          222.6 Hz         4492 us
|   |__ 3) vehicle_air_data               74.3 Hz        13463 us
|   |__ 4) vehicle_imu                   202.4 Hz         4941 us
|   |__ 5) ekf2                          200.3 Hz         4993 us
|   |__ 6) mc_pos_control                 50.1 Hz        19956 us
|   \__ 7) land_detector                  50.0 Hz        19987 us (20000 us)
|__ 7) wq:rate_ctrl
|   |__ 1) vehicle_angular_velocity      400.6 Hz         2496 us
|   \__ 2) mc_rate_control               400.5 Hz         2497 us
\__ 8) wq:attitude_ctrl
    \__ 1) mc_att_control                200.3 Hz         4992 us
nsh>
```

**Comment by bys1123** [comment]
![image](https://user-images.githubusercontent.com/4748411/83543670-5832ed80-a52f-11ea-87fa-1bb8d9e6c10b.png)
Thanks for testing, We sure we are using V3 bootloader

**Comment by davids5** [comment]
Is this just one unit or on many?

**Comment by bys1123** [comment]
@davids5 many, include Holybro current produce Pixhawk Mini.

**Comment by davids5** [comment]
@bys1123 can you get one sent to me?

**Comment by bys1123** [comment]
@davids5 No problem, I'll let Holybro send you one.

**Comment by bys1123** [comment]
Fixed by #15013

**Comment by dagar** [comment]
Thanks for the quick feedback.
